client/my-sites/pages/page-list.jsx

```diff
@@ -243,7 +243,7 @@ var Pages = React.createClass({
 
 		const site = this.props.sites.getSelectedSite();
 
-		if ( site && site.options && site.options.show_on_front === 'posts' ) {
+		if ( this.props.status === 'published' && site && site.options && site.options.show_on_front === 'posts' ) {
```

Comment:
would this be a fitting time to bring in a helper to clean up the nested data get?

```js
import get from 'lodash/get';

if ( this.props.status === 'published' && get( site, 'options.show_on_front' ) === 'posts' ) {
	…
}
```

or alternatively

```js
import matchesProperty from 'lodash/matchesProperty';

if ( this.props.status === 'published' && matchesProperty( 'options.show_on_front', 'posts' )( site ) ) {
```